File: ALDS1/1_1_D_Maximum_Profit.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> r(n);

  cin >> r.at(0);
  int minv = r.at(0);
  int maxv = -1 * pow(10, 9);

  for (int i = 1; i < n; i++) {
    cin >> r.at(i);
    maxv = max(maxv, r[i] - minv);
    minv = min(minv, r[i]);
  }
  
  cout << maxv << endl;
}

File: ALDS1/Chapter2/1_1_D_Maximum_Profit.py
R = [int(input()) for _ in [0]*int(input())]
minv = R[0]
maxv = -1e+10
for j in range(1, len(R)):
    maxv = max(maxv, R[j]-minv)
    minv = min(minv, R[j])
print(maxv)
"""
R = [int(input()) for _ in [0]*int(input())]
print(max([max(R[i+1:]) - R[i] for i in range(len(R)-1)]))
"""
"""
R = [int(input()) for _ in [0]*int(input())]
maxv = -1e+10
for j in range(1, len(R)-1):
    for i in range(0, j-1):
        maxv = max(maxv, R[j]-R[i])
print(maxv)
"""